xarray/tests/test_formatting.py

```diff
@@ -218,31 +218,68 @@ def test_attribute_repr(self) -> None:
         assert "\n" not in newlines
         assert "\t" not in tabs
 
-    def test_index_repr(self):
+    def test_index_repr(self) -> None:
         from xarray.core.indexes import Index
 
         class CustomIndex(Index):
-            def __init__(self, names):
+            names: tuple[str, ...]
+
+            def __init__(self, names: tuple[str, ...]):
                 self.names = names
 
             def __repr__(self):
                 return f"CustomIndex(coords={self.names})"
 
-        coord_names = ["x", "y"]
+        coord_names = ("x", "y")
         index = CustomIndex(coord_names)
-        name = "x"
+        names = ("x",)
 
-        normal = formatting.summarize_index(name, index, col_width=20)
-        assert name in normal
+        normal = formatting.summarize_index(names, index, col_width=20)
+        assert names[0] in normal
+        assert len(normal.splitlines()) == len(names)
         assert "CustomIndex" in normal
 
-        CustomIndex._repr_inline_ = (
-            lambda self, max_width: f"CustomIndex[{', '.join(self.names)}]"
+        CustomIndex._repr_inline_ = (  # type: ignore
+            lambda self, max_width: f"CustomIndex[{', '.join(self.names)}]"  # type: ignore
```

Comment:
I don't understand why we do this monkeypatching? 
Can't we just make a `CustomIndex2` with that change?

Reply:
Yeah, good idea.
And if not, I wouldn't worry too much about typing in patches, that usually doesn't work nicely.

Reply:
good point. The idea was to "try the same thing, but this time with `_repr_inline_` defined", which is not quite as easy to understand if we define a separate (sub-)class. Not sure how important the slight increase in readability is compared to the typing issues, though.

Reply:
I prefer a new class at least, it's the more common code pattern and therefore a little easier to read and understand.

```python
class CustomIndexReprInline(CustomIndex):
    def _repr_inline_(self, max_width):
        return f"CustomIndex[{', '.join(self.names)}]"
```